contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3833,6 +3882,52 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 					return;
 				}
 			}
+			/*
+			 * Set current user to bbf_role_admin while restoring babelfish roles/grants
+			 * so that it becomes their admin/GRANTOR. We will do this only if the current
+			 * user is superuser since only superuser is allowed to perform dump/restore.
+			 * Note that no additional permission checks are needed as superusers can
+			 * anyway perform this action.
+			 */
+			else if (!IsBinaryUpgrade && babelfish_dump_restore && superuser())
+			{
+				Oid 			save_userid;
+				int 			save_sec_context;
+				ListCell		*item;
+				bool			isadmin = false;
+				GrantRoleStmt	*stmt = (GrantRoleStmt *) parsetree;
+
+				foreach(item, stmt->opt)
+				{
+					DefElem    *opt = (DefElem *) lfirst(item);
+					char	   *optval = defGetString(opt);
+
+					if (strcmp(opt->defname, "admin") == 0)
+					{
+						parse_bool(optval, &isadmin);
+						break;
+					}
+				}
+				if (isadmin)
+					break;
```

Comment:
Please explain this.

Reply:
Added comment. Basically, bbf_role_admin can't be grantor of it's own admin grants.